src/main/java/org/wildfly/security/auth/server/IdentityLocator.java

```diff
@@ -136,27 +148,31 @@ public static IdentityLocator fromName(String name) {
     public static IdentityLocator fromEvidence(Evidence evidence) {
         Assert.checkNotNullParam("evidence", evidence);
         final Principal principal = evidence.getPrincipal();
-        return new IdentityLocator(principal instanceof NamePrincipal ? principal.getName() : null, principal, evidence);
+        return new IdentityLocator(null, principal instanceof NamePrincipal ? principal.getName() : null, evidence);
     }
 
-    /**
-     * Shortcut method to construct an identity locator from just a principal instance.  The name will be populated from
-     * the principal if it is an instance of {@code NamePrincipal}.
-     *
-     * @param principal the principal (must not be {@code null})
-     * @return the identity locator (not {@code null})
-     */
-    public static IdentityLocator fromPrincipal(Principal principal) {
-        Assert.checkNotNullParam("principal", principal);
-        return new IdentityLocator(principal instanceof NamePrincipal ? principal.getName() : null, principal, null);
+    @Override
```

Comment:
As long as this class has evidence, it shouldn't have equals/hashCode, even if evidence is excluded from hashCode.